security/advancedtls/provider.go

```diff
@@ -0,0 +1,184 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package advancedtls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"google.golang.org/grpc/credentials/tls/certprovider"
+)
+
+const (
+	defaultTestTimeout = 1 * time.Second
+)
+
+// IdentityPemFileProviderOptions contains fields to be filled out by users
+// for obtaining identity credential files.
+type IdentityPemFileProviderOptions struct {
+	CertFile string
+	KeyFile  string
+	Interval time.Duration
+}
+
+// IdentityPemFileProvider keeps identity credential implementations up to
```

Comment:
We can rephrase this to "IdentityPemFileProvider implements certprovider.Provider. It provides the most up-to-date identity private key and certificates based on the  input PEM files.". Something like that.

Reply:
Fixed!